include/deal.II/lac/precondition.h

```diff
@@ -1975,7 +1975,8 @@ class PreconditionChebyshev : public Subscriptor
                    const double              eig_cg_residual     = 1e-2,
                    const double              max_eigenvalue      = 1,
                    const EigenvalueAlgorithm eigenvalue_algorithm =
-                     EigenvalueAlgorithm::lanczos);
+                     EigenvalueAlgorithm::lanczos,
+                   const bool use_fourth_kind = false);
```

Comment:
I wonder if we should already provision for other possibilities, in which case it would make more sense to have an enum that can be extended in the future. Similarly to the setting for the Lanczos / power iteration eigenvalue iteration. I would call them
```c++
enum PolynomialType
{
  first_kind,
  fourth_kind
};
```
We would then also need to augment the documentation, as the fourth kind polynomials do not respect a lower bound (and in fact, I would not know how to use them as a direct solver because the error estimate is different).